### PR TITLE
New Tag Manager additions

### PR DESCRIPTION
**Pull request by npinciak**
Apologies for duplicate pull requests. Issues from previous pull should be resolved

Yahoo! Tag Manager
- https://www.shop.post.japanpost.jp
- http://skynet-c.jp/
- http://www.jal.co.jp/
- http://www.itscom.net/
- http://n.shop.vector.co.jp/
- https://learning.sankei.co.jp/
- http://www.sekaimon.com/
- https://www.ikyu.com/
- https://www.skincare-univ.com/
- https://www.subaru.jp/

Ensighten
- https://www.gymboree.com/
- https://www.freecreditscore.com/
- https://www.wcvb.com/
- https://www.wesh.com/
- https://www.wgal.com/
- http://www.wnba.com/
- https://www.fossil.com/us/en.html
- https://www.wdsu.com/
- https://www.sfgate.com/

Signal
- https://www.pavilions.com/
- https://www.vikingdirect.ie/en/
- https://www.officedepot.com/
- http://www.shopping.com/
- https://www.seattletimes.com/
- https://www.safeway.com/

Adobe DTM
- https://www.verticalresponse.com/
- https://www.golfdigest.com/ 
- https://www.geotrust.com/
- https://www.webmd.com/
- https://www.fandango.com/
- https://www.microsoft.com/en-us/windows
- https://www.fedex.com
- https://www.glennbeck.com/
- https://www.ford.com/
- https://www.panerabread.com/

### CODE REVIEW

File: src/apps.json [hunk 1]
```diff
@@ -11126,6 +11154,15 @@
       },
       "website": "http://smallbusiness.yahoo.com/ecommerce"
     },
+    "Yahoo! Tag Manager": {
+      "cats": [
+        42
+      ],
+      "html":"<!-- (?:End )?Yahoo! Tag Manager -->",
+      "script": "b\\.yjtag\\.jp/iframe",
```
Comment (jvoisin) [inline src/apps.json]: I think that the `.jp` extension is specific to Japan.

File: src/apps.json [hunk 2]
```diff
@@ -9606,8 +9633,9 @@
       },
       "script": [
         "^//tags\\.tiqcdn\\.com/",
-        "/tealium/utag\\.js$"
-      ],
+        "/tealium/utag\\.js$",
+        "/utag\\.js"
```
Comment (jvoisin) [inline src/apps.json]: This is a bit too false-positive prone.
Moreover, the line above is already detecting `utag.js`, under the `tealium` folder: this should be enough.

File: src/apps.json [hunk 3]
```diff
@@ -8833,6 +8849,17 @@
       ],
       "website": "http://shopware.com"
     },
+    "Signal": {
+      "cats": [
+        42
```
Comment (jvoisin) [inline src/apps.json]: This is an "Identity Resolution & Data Onboarding | Real-Time Marketing" tool, not a "tag manager" one.

File: src/apps.json [hunk 4]
```diff
@@ -8833,6 +8849,17 @@
...
+      ],
```
Comment (jvoisin) [inline src/apps.json]: Please add `signalData` to the `js` detection method too :)

File: src/apps.json [hunk 5]
```diff
@@ -2819,6 +2827,14 @@
       },
       "website": "http://emberjs.com"
     },
+    "Ensighten": {
+      "cats": [
+        42
+      ],
+      "script": "^\"|//nexus\\.ensighten\\.com/.*/bootstrap\\.js",
```
Comment (jvoisin) [inline src/apps.json]: How about `^//nexus.ensighten.com/` instead?

File: src/apps.json [hunk 6]
```diff
@@ -277,6 +277,14 @@
       "url": "\\.cfm(?:$|\\?)",
       "website": "http://adobe.com/products/coldfusion-family.html"
     },
+    "Adobe DTM": {
+      "cats": [
+        42
+      ],
+      "script": "\"//assets\\.adobedtm\\.com",
```
Comment (jvoisin) [inline src/apps.json]: What about `"^//assets\\.adobedtm\\.com"` instead?

File: src/apps.json (hunk 2 above)
Comment (npinciak) [reply src/apps.json]: From what I have observed, not all websites use the /tealium/ path to host this file. :)

For example: 
- https://www.newbalance.com/
  - Has their tag located here: //tags.tiqcdn.com/utag/newbalance/nb-us/prod/utag.js
  - With the current Tealium detection, it does not detectable here
- https://www.imotors.com/
  - Has their tag located here: //tags.tiqcdn.com/utag/one-planet-ops/auto/prod/utag.js
  - With the current Tealium detection, it does not detectable here

File: src/apps.json (hunk 3 above)
Comment (npinciak) [reply src/apps.json]: Do we have a category number for this?

File: src/apps.json (hunk 2 above)
Comment (jvoisin) [reply src/apps.json]: Sure, but `utag.js` is way too short (this kind of name could be generated by a javascript minimizer) to use it as a viable mean of fingerprinting.

File: src/apps.json (hunk 1 above)
Comment (jvoisin) [reply src/apps.json]: You're absolutely right, my bad :)

File: src/apps.json (hunk 5 above)
Comment (npinciak) [reply src/apps.json]: Thanks, updated to 

    //nexus.ensighten.com/

File: src/apps.json (hunk 6 above)
Comment (npinciak) [reply src/apps.json]: Thank you change has been made 👍

File: src/apps.json (hunk 1 above)
Comment (npinciak) [reply src/apps.json]: Not a problem 👍

File: src/apps.json (hunk 3 above)
Comment (jvoisin) [reply src/apps.json]: Maybe `32` or `10`

File: src/apps.json [hunk 7]
```diff
@@ -2171,7 +2179,10 @@
         "criteo_pubtag": "",
         "criteo_q": ""
       },
-      "script": "//(?:cas\\.criteo\\.com|(?:[^/]\\.)?criteo\\.net)/",
+      "script":[ 
+        "//(?:cas\\.criteo\\.com|(?:[^/]\\.)?criteo\\.net)/",
+        "//static.criteo.net/js/ld/ld.js"
```
Comment (jvoisin) [inline src/apps.json]: I think that `"\.criteo\.(?:com|net)/"` should be enough :)

File: src/apps.json [hunk 8]
```diff
@@ -2819,6 +2830,14 @@
       },
       "website": "http://emberjs.com"
     },
+    "Ensighten": {
+      "cats": [
+        42
+      ],
+      "script": "//nexus.ensighten.com/",
```
Comment (jvoisin) [inline src/apps.json]: Don't forget to escape the dots ;)

File: src/apps.json (hunk 7 above)
Comment (npinciak) [reply src/apps.json]: Thanks for the input here. This is not a valid script we use at Criteo (I am a dev there :) ). The following script is a mandatory script used to call all Criteo functions, and is present with all tracking.

    "//static.criteo.net/js/ld/ld.js"

Comment (jvoisin) [reply src/apps.json]: I think that considering any javascript file served from a criteo domain as an indicator of criteo's presence isn't unreasonable. Moreover, this future-proofs at bit the signature :)

Comment (npinciak) [reply src/apps.json]: Fair enough! :)

Thanks!

**Comment by AliasIO** [comment]
Thank you!

**Pull request merged**